querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathInits.java

```diff
@@ -93,4 +94,18 @@ public boolean isInitialized(String property) {
         return initAllProps || propertyToInits.containsKey(property);
     }
 
+    public static PathInits getFor(PathMetadata metadata, PathInits root) {
+        if (metadata.isRoot()) {
+            return root;
+        } else if (metadata.getParent().getMetadata().isRoot()) {
+            return DIRECT;
+        } else {
+            return DEFAULT;
+        }
+    }
+
+    public static PathInits getFor(PathMetadata metadata) {
+        return getFor(metadata, DIRECT2);
+    }
```

Comment:
Question: what's this one used for?
Or are you planning to use this one for something?